### PR TITLE
KT-54405: K2 JVM Fix wrong action in val declarations conflicting 

### PR DESCRIPTION
**Pull request by lfkdsk**
K2 compiler allows val redeclaration, Fixes: https://youtrack.jetbrains.com/issue/KT-54405/K2-compiler-allows-val-redeclaration

``` kotlin
fun main(args: Array<String>) {

    val someVal = "One" //Highlighted with e: [REDECLARATION] Conflicting declarations: val someVal: String, val someVal: Int
    val someVal = 42//Highlighted with e: [REDECLARATION] Conflicting declarations: val someVal: String, val someVal: Int

    println (someVal)
}
```

### CODE REVIEW

File: compiler/fir/checkers/src/org/jetbrains/kotlin/fir/analysis/checkers/declaration/FirConflictsChecker.kt [hunk 1]
```diff
@@ -322,6 +329,22 @@ object FirConflictsChecker : FirBasicDeclarationChecker() {
         }
     }
 
+    private fun checkBlock(body: FirBlock, inspector: DeclarationInspector) {
```
Comment (KvanTTT) [inline compiler/fir/checkers/src/org/jetbrains/kotlin/fir/analysis/checkers/declaration/FirConflictsChecker.kt:332]: The method doesn't check in the following case:

```kt
fun main(b: Boolean) {
    if (b) {
        val someVal2 = 2
        val someVal2 = "Two"
    }
}
```

Please consider more cases.

**Review by KvanTTT — CHANGES_REQUESTED** [review]
Thanks for the contribution!

But the header of this PR is confusing. This redeclaration is not JVM-specific diagnostics, please remove `JVM`.

Also, please fix other remarks.

**Comment by KvanTTT** [comment]
Also, there are a lot of failed tests (at least `FirOldFrontendDiagnosticsTestGenerated`). Please fix them, they can run locally.

**Comment by lfkdsk** [comment]
Fix test and fix other original testData result.

**Comment by lfkdsk** [comment]
@KvanTTT

**Comment by KvanTTT** [comment]
Thank you for the contribution, it will be merged eventually later. Sorry, but I'm busy with more priority tasks for now.

**Review by KvanTTT — COMMENTED** [review]
Thanks for the contribution but this bug was eventually fixed by https://github.com/JetBrains/kotlin/commit/e49bb1fe37950a540e312bb6be020e46af8dcc38

**Pull request closed without merging**